### DISCUSSION TITLE
Is there a way to disable reactivity?

### DISCUSSION THREAD

**Discussion by elite174** [solidjs/solid#1063] [Q&A] [answered]
Hello guys!
I have a component which uses reactive getters to update DOM. It's a complex component representing kind of visual board. I receive updates on the web sockets to update the visual state of the board. On this board I can move and resize elements. When a user does some interaction (move or resize) I need this component to stop being reactive: so if I got some update on the web sockets I don't need to update the DOM because interaction is in progress. How can I achieve this behavior?

I made simple component trying to disable reactivity, but it doesn't work. Is it possible?

```
import { render } from "solid-js/web";
import {
  createSignal,
  Component,
  untrack,
  Accessor,
  onMount,
  createEffect,
  ParentComponent,
  Show,
} from "solid-js";

const A: Component<{ a: number }> = (props) => {
  console.log("inited");

  onMount(() => console.log("mount"));
  return <div>A: {props.a}</div>;
};

function Counter() {
  const [a, setA] = createSignal(0);

  const increment = () => setA((v) => v + 1);

  return untrack(() => (
    <div>
      <button onClick={increment}>click</button>
      <div>
        <A a={a()} />
      </div>
    </div>
  ));
}

render(() => <Counter />, document.getElementById("app")!);
```

**Comment by ryansolid** [comment, marked as answer]
Yeah reactivity is nested so an `untrack` there will untrack any top-level read but a nested expression will be in its own context. The important part is that signals are designed to keep things in sync. It's possible to opt-out (Ie don't use a signal, access it outside the expression, `untrack`) but its not possible to temporarily opt-out unless that mechanism uses signals itself since signal and the DOM would be out of sync, which is a state that Solid can't make guarantees about when it resumed.

So my thinking would be:
1. Do I need this to resume reactivity after? If so opting out of reactivity doesn't make sense I'd use a conditional signal to disconnect the source from the output. So it's still reactive but stops updating temporarily.
2. If I don't need to resume I might consider managing the lifecycle of that part of the app with `createRoot` and handle disposal myself. Just call dispose and then everything below that loses reactivity.

**Reply by elite174** [reply]
Thank you, I got the point. Yeah, I thought in the similar way, but I hoped there is another way...

**Comment by atk** [comment]
I guess what you actually want is a derived reactive state that only forwards changes if a condition is met. I see two ways to handle this:

1. Use a createMemo with equals option that also checks if the condition is met.
2. Use a separate signal and update inside an effect if the condition is not met. This gives you the opportunity to update from the last value once the updates should resume.
